crates/driver/src/domain/competition/auction.rs

```diff
@@ -41,6 +42,7 @@ impl Auction {
         tokens: impl Iterator<Item = Token>,
         deadline: time::Deadline,
         eth: &Ethereum,
+        surplus_capturing_jit_order_owners: &HashSet<eth::Address>,
```

Comment:
There is still currently one instance where you create a new `HashSet` only to pass the reference and have this function clone the reference to the newly created `HashSet` anyway.
```suggestion
        surplus_capturing_jit_order_owners: HashSet<eth::Address>,
```